Marketplace/guitar/guitar.js

```diff
@@ -0,0 +1,666 @@
+    var injectorL;
+    var injectorR;
+    var timeout = null;
+
+    var Guitar = function() {
+        _this = this;
+    };
+    Guitar.prototype = {
+        preload: function(entityID) {
+            _this.entityID = entityID;
+            Overlays.mousePressOnOverlay.connect(this.mousePress);
+            Entities.findEntities(_this.entityID, SEARCH_RADIUS).forEach(function(element) {
+                var name = Entities.getEntityProperties(element, 'name').name;
+                if (name === "Guitar Amp Left CC-BY Poly by Google") {
+                    ampLeft = element;
+                    ampLeftPosition = Entities.getEntityProperties(element, 'position').position;
+                } else if (name === "Guitar Amp Right CC-BY Poly by Google") {
+                    ampRight = element;
+                    ampRightPosition = Entities.getEntityProperties(element, 'position').position;
+                } else if ((name.indexOf("Guitar Body") !== NOT_FOUND) || (name.indexOf("Guitar Neck") !== NOT_FOUND)) {
+                    colors.push(element);
+                }
+            });
+            for ( var i =1; i <= 8; i++){
+                sounds["Red" + i + "L"] = SoundCache.getSound(Script.resolvePath("sounds/red/" + i + "L.wav"));
+                sounds["Red" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/red/" + i + "R.wav"));
+                sounds["Orange" + i + "L"] = SoundCache.getSound(Script.resolvePath("sounds/orange/" + i + "L.wav"));
+                sounds["Orange" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/orange/" + i + "R.wav"));
+                sounds["Yellow" + i + "L"] = SoundCache.getSound(Script.resolvePath("sounds/yellow/" + i + "L.wav"));
+                sounds["Yellow" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/yellow/" + i + "R.wav"));
+                sounds["Green" + i + "L"] = SoundCache.getSound(Script.resolvePath("sounds/green/" + i + "L.wav"));
+                sounds["Green" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/green/" + i + "R.wav"));
+                sounds["Blue" + i + "L"] = SoundCache.getSound(Script.resolvePath("sounds/blue/" + i + "L.wav"));
+                sounds["Blue" + i + "R"] = SoundCache.getSound(Script.resolvePath("sounds/blue/" + i + "R.wav"));
+            }
+        },
+        mousePress: function(id, event) {
+            if (overlays.indexOf(id) !== NOT_FOUND) {
+                if (timeout) {
+                    Script.clearTimeout(timeout);
+                    timeout = null;
+                }
+                timeout = Script.setTimeout(function(){
+                    _this.releaseGrab();
+                    timeout = null;
+                }, STILL_PLAYING_TIMEOUT_MS);
+                if (ampCheckInterval) {
+                    Script.clearTimeout(ampCheckInterval);
+                }
+                ampCheckInterval = Script.setInterval(this.updateAmpPositions, UPDATE_POSITION_MS);
```

Comment:
I would move the amp check into startGrab but not set it as an interval - I think we should assume that the amps will not be moving during play, so it would be fine to update the position at the start of grabbing the guitar instead of on an interval.